cheatsheets/Nodejs_security_cheat_sheet.md

```diff
@@ -0,0 +1,371 @@
+
+```JavaScript
+app.use(express.urlencoded({ limit: “1kb” }));
+app.use(express.json({ limit: “1kb” }));
+app.use(express.multipart({ limit:”10mb” }));
+app.use(express.limit(“5kb”)); // this will be valid for every other content type
+```
+
+## Use strict mode
+
+JavaScript has some unsafe features that lies within the language itself. In order to remove these features, ES5 included a strict mode for developers. With this mode, errors that were silent previously are thrown. It also restricted optimizations for JavaScript engines. With strict mode, previously accepted bad syntax causes real errors. use "use strict". Because of these improvements, you should always use strict mode in your application. In order to enable strict mode, you just need to write _‘use strict’;_ on top of your code.
+
+## Use object property descriptors
+
+Object properties include 3 hidden attributes: writable (if false, property value cannot be changed), enumerable (if false, property cannot be used in for loops) and configurable (if false, property cannot be deleted). When defining an object property through assignment, these three hidden attributes are set to true by default. These properties can be set as follows:
+
+```JavaScript
+var o = {};
+Object.defineProperty(o, “a”, {
+    writable: true,
+    enumerable: true,
+    configurable: true,
+    value: “A”
+});
+```
+
+Apart from these, there are some special functions for object attributes. Object.preventExtensions() prevents new properties from being added to the object.
+
+## Handle errors in asynchronous calls
+
+Errors that occur within asynchronous callbacks can be easily forgettable. Therefore, as a general principle first argument to the asynchronous calls should be an Error object. Also, express routes handle errors itself, but it should be always remembered that errors occurred in asynchronous calls made within express routes are not handled, unless an Error object is sent as a first argument.
+
+Errors in these callbacks can be propagated as many times as possible. Each callback that the error has been propagated to can ignore, handle or propagate the error.
+
+## Listen to errors when using EventEmitter
+
+When using EventEmitter, errors can occur anywhere in the event chain. Normally, if an error occurs in an EventEmitter object, an error event with an Error object as its argument is called. However, if there are no attached listeners to that error event, the Error object that is sent as argument is thrown and becomes an uncaught exception. In short, if you do not handle errors within an EventEmitter object properly, these unhandled errors may crash your application. Therefore, you should always listen to error events when using EventEmitter objects.
```

Comment:
Second sentence is quite hard to read.

`as an argument` instead of `as argument`

Reply:
changed accordingly